saleor/graphql/order/bulk_mutations/order_bulk_create.py

```diff
@@ -1233,178 +1576,248 @@ def create_single_fulfillment(
                 quantity=line_input["quantity"],
             )
             lines.append(OrderBulkFulfillmentLine(fulfillment_line, warehouse))
+            line_index += 1
 
         return OrderBulkFulfillment(fulfillment=fulfillment, lines=lines)
 
     @classmethod
     def create_single_order(
         cls, order_input, object_storage: Dict[str, Any]
-    ) -> OrderWithErrors:
-        order = OrderWithErrors()
-        cls.validate_order_input(order_input, order.order_errors)
-        order_instance = Order(currency=order_input["currency"])
+    ) -> OrderBulkCreateData:
+        order_data = OrderBulkCreateData()
+        cls.validate_order_input(order_input, order_data, object_storage)
+        if order_data.is_critical_error:
+            return order_data
 
+        order_data.order = Order(currency=order_input["currency"])
         # get order related instances
-        instances = cls.get_instances_related_to_order(
+        cls.get_instances_related_to_order(
             order_input=order_input,
-            errors=order.order_errors,
+            order_data=order_data,
             object_storage=object_storage,
         )
         delivery_input = order_input["delivery_method"]
         delivery_method = cls.get_delivery_method(
```

Comment:
Just to confirm: Are you taking into account that some of the orders would not require a shipping? Right?

Reply:
Good spot, I omitted this case. Implemented in last push.